### ISSUE TITLE
Integrate Qdrant In-Memory Storage into LangChain

### ISSUE THREAD

**Issue by slavakurilyak** [Significant-Gravitas/Auto-GPT#273]
I propose integrating [Qdrant](https://github.com/qdrant/qdrant-client)'s in-memory storage capabilities into LangChain to enable running Auto-GPT in local mode without requiring a server and enhance the memory feature of Auto-GPT.

Qdrant is an open-source vector search engine and vector database with in-memory payload storage for fast search speeds. By integrating Qdrant's in-memory storage into LangChain, we can enable local mode for Auto-GPT without the need for a server. This would not only benefit developers who want to run Auto-GPT locally without incurring server costs but also enhance the memory feature of Auto-GPT.

**Comment by slavakurilyak** [comment]
@davelm42 Perhaps a new feature?

**Comment by davelm42** [comment]
Yea, I think that's what I'm going to work on this weekend unless someone beats me to it.

**Comment by dahifi** [comment]
I was going to suggest Weaviate, Llama-index/gpt-index already supports it -- I am unfamiliar with these others.  

I am starting a local branch to add the Weaviate client, I'll abstract the memory interface and see if I can make it work without breaking anything.

**Comment by dahifi** [comment]
This appears to be a sample of the current langchain vector stores:
```
    "ElasticVectorSearch",
    "FAISS",
    "VectorStore",
    "Pinecone",
    "Weaviate",
    "Qdrant",
    "Milvus",
    "Chroma",
    ```

**Comment by davelm42** [comment]
Looks like someone has a Weaviate PR out there right now: 

https://github.com/Torantulino/Auto-GPT/pull/424

**Comment by Pwuts** [comment]
Closing as resolved with #424. Please reopen if you disagree.
